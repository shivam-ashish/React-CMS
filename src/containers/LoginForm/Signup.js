import React, { Component } from 'react';
import firebase from 'firebase';
import classes from './LoginForm.module.css';
import Fire from '../../config/fire';
import MDSpinner from 'react-md-spinner';

class Signup extends Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
    this.signup = this.signup.bind(this);
    this.state = {
      email: '',
      password: '',
      name: '',
      spinner: '',
    };
  }

  signup = (e) => {
    const { email, password } = this.state;
    this.setState({ spinner: true });
    e.preventDefault();
    Fire.auth().createUserWithEmailAndPassword(email, password)
      .then((u) => {
        const { name } = this.state;
        const database = firebase.database();
        const ref = database.ref('users');
        const data = {
          userName: name,
        };
        ref.push(data);
      })
      .catch((error) => {
        alert(error.message);
        this.setState({ spinner: false });
      });
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  render() {
    const { email, password, name, spinner } = this.state;
    return (
      <div className={classes.loginForm}>
        <h1>Sign UP Here</h1>
        <label htmlFor="email">Name : </label>
        <input value={name} onChange={this.handleChange} type="text" name="name" />
        <label htmlFor="email">Email Address : </label>
        <input value={email} onChange={this.handleChange} type="email" name="email" />
        <label htmlFor="email">Password : </label>
        <input value={password} onChange={this.handleChange} type="password" name="password" />
        <button type="submit" className={classes.loginBtn} onClick={this.props.toggle}>Back to Login</button>
        <button type="submit" className={classes.loginBtn} onClick={(e) => this.signup(e)}>{spinner?<MDSpinner/>:'Sign Up'}</button>
      </div>
    );
  }
}

export default Signup;
