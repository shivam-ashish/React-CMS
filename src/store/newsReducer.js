const initialState = {
  news: null,
};

const newsReducer = (state = initialState, { payload }) => {
  return {
    ...state,
    news: payload,
  };
};

export default newsReducer;
