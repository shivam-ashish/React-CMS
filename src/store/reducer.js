
const initialState ={
    isLoggedIn: false,
    user: 'user1',
}

const reducer = ( state = initialState, action ) => {
    const newState = { ...state };
    switch (action.type){
    case "LOGIN_STATE":
    return{
        ...state,
        isLoggedIn: Boolean,
        history: state.history.concat({
            isLoggedIn: Boolean
        })
    };
    case "UPDATE_USER":
    return{
        ...state,
        
        history: state.history.concat({
            
        })
    };
    default:
    return newState;
}
};

export default reducer;