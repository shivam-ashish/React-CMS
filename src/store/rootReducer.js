import { createStore, combineReducers } from 'redux';
import reducer from './reducers/reducer';
import newsReducer from './reducers/newsReducer';
import blogReducer from './reducers/blogReducer';

const rootReducer = combineReducers({
  reducer,
  newsReducer,
  blogReducer,
});

const store = createStore(rootReducer);

export default store;
