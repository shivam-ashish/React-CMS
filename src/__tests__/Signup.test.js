import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Signup from '../containers/LoginForm/Signup';
import store from '../store/rootReducer';

let wrapped;

beforeEach(() => {
  wrapped = mount(<Provider store={store}><Router><Signup props="xyz" /></Router></Provider>);
});
afterEach(() => {
  wrapped.unmount();
});

it('has a AddEdit component structure', () => {
  expect(wrapped.find('div').length).toEqual(2);
  expect(wrapped.find('h1').length).toEqual(1);
  expect(wrapped.find('input').length).toEqual(3);
  expect(wrapped.find('label').length).toEqual(3);
  expect(wrapped.find('Button').length).toEqual(2);
});

it('has text fields where users can type in', () => {
  wrapped.find('#name').simulate('change', {
    target: { value: 'username', name: 'name' },
  });
  wrapped.find('#email').simulate('change', {
    target: { value: 'email address', name: 'email' },
  });
  wrapped.find('#password').simulate('change', {
    target: { value: 'password', name: 'password' },
  });
  wrapped.update();
  expect(wrapped.find('#name').prop('value')).toEqual('username');
  expect(wrapped.find('#email').prop('value')).toEqual('email address');
  expect(wrapped.find('#password').prop('value')).toEqual('password');
});

it('it matches the snapshot', () => {
  const component = create(wrapped);
  expect(component.toJSON()).toMatchSnapshot();
});
