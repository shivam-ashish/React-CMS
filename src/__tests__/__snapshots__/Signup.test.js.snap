// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches the snapshot 1`] = `
<div
  className="loginForm"
>
  <h1
    className="heading"
  >
    SignUp Page
  </h1>
  <label
    className="labelField"
    htmlFor="email"
  >
    Name : 
  </label>
  <input
    className="inputField"
    id="name"
    name="name"
    onChange={[Function]}
    type="text"
    value=""
  />
  <label
    className="labelField"
    htmlFor="email"
  >
    Email Address : 
  </label>
  <input
    className="inputField"
    id="email"
    name="email"
    onChange={[Function]}
    type="email"
    value=""
  />
  <label
    className="labelField"
    htmlFor="email"
  >
    Password : 
  </label>
  <input
    className="inputField"
    id="password"
    name="password"
    onChange={[Function]}
    type="password"
    value=""
  />
  <button
    className="login"
    type="Back to Login"
  >
    Back to Login
  </button>
  <button
    className="signup"
    click={[Function]}
    onClick={[Function]}
    type="SignUp"
  >
    <div>
      âœ”
    </div>
  </button>
</div>
`;
